Nullable Timestamptz unsupported with now

## Setup

### Versions

- **Rust:**
nightly
- **Diesel:**
0.14.1
- **Database:**
postgres
- **Operating System**
Fedora

### Feature Flags

- **diesel:**
- **diesel_codegen:**

## Problem Description

I was attempting to write a query that updated a timestamp column that was nullable and received a compile error. Example query:

```rust
diesel::update(import_jobs.filter(completed_at.is_null()))
        .set(completed_at.eq(diesel::expression::dsl::now))
        .execute(conn)?;
```

### What are you trying to accomplish?
^^^

### What is the expected output?
Successful compile

### What is the actual output?

```rust
error[E0271]: type mismatch resolving `<diesel::expression::nullable::Nullable<diesel::expression::now> as diesel::Expression>::SqlType == diesel::types::Nullable<diesel::types::Timestamptz>` --> src/importer.rs:52:25 | 52 | .set((failed_at.eq(diesel::expression::dsl::now.nullable()), failed_reason.eq("aborted"))) | ^^ expected struct `diesel::types::Timestamp`, found struct `diesel::types::Timestamptz` | = note: expected type `diesel::types::Nullable<diesel::types::Timestamp>` found type `diesel::types::Nullable<diesel::types::Timestamptz>`
```
## Current work around

@sgrif Recommended this for now:

```rust
let nullable_timestamptz = AsExpression::<Timestamptz>::as_expression(diesel::expression::dsl::now).nullable();

diesel::update(import_jobs.filter(completed_at.is_null()))
        .set(completed_at.eq(nullable_timestamptz))
        .execute(conn)?;
```

--- a/diesel/src/expression/functions/date_and_time.rs
+++ b/diesel/src/expression/functions/date_and_time.rs
@@ -69,3 +69,12 @@ impl AsExpression<Timestamptz> for now {
         Coerce::new(self)
     }
 }
+
+#[cfg(feature="postgres")]
+impl AsExpression<Nullable<Timestamptz>> for now {
+    type Expression = Coerce<now, Nullable<Timestamptz>>;
+
+    fn as_expression(self) -> Self::Expression {
+        Coerce::new(self)
+    }
+}

--- a/diesel_tests/tests/expressions/date_and_time.rs
+++ b/diesel_tests/tests/expressions/date_and_time.rs
@@ -2,6 +2,7 @@ use schema::{connection, TestConnection};
 use diesel::*;
 use diesel::data_types::*;
 use diesel::expression::dsl::*;
+use diesel::types::Nullable;
 
 table! {
     has_timestamps {
@@ -53,6 +54,25 @@ fn now_can_be_used_as_timestamptz() {
                         (NOW() - '1 day'::interval)").unwrap();
 
     let created_at_tz = sql::<Timestamptz>("created_at");
+    let before_now = has_timestamps.select(id)
+        .filter(created_at_tz.lt(now))
+        .load::<i32>(&connection);
+    assert_eq!(Ok(vec![1]), before_now);
+}
+
+#[test]
+#[cfg(feature = "postgres")]
+// FIXME: Replace this with an actual timestamptz expression
+fn now_can_be_used_as_nullable_timestamptz() {
+    use self::has_timestamps::dsl::*;
+    use diesel::types::Timestamptz;
+
+    let connection = connection();
+    setup_test_table(&connection);
+    connection.execute("INSERT INTO has_timestamps (created_at) VALUES \
+                        (NOW() - '1 day'::interval)").unwrap();
+
+    let created_at_tz = sql::<Nullable<Timestamptz>>("created_at");
     let before_now = has_timestamps.select(id)
         .filter(created_at_tz.lt(now))
         .load::<i32>(&connection);

